### PR TITLE
Adding Activation Code UI

### PR DESCRIPTION
**Pull request by Emily-ejag**
## Proposed changes
Adding activation code button to the ListOrgs component, then a modal will show asking the users to copy the code or URL

Steps:

1. Go to the `List Organizations `page, and click on the `Invitation Link` column click on `Invite Users`
2. With this link go to a different tab, and change the roar.education part for the pr link part (or if localhost, switch to localhost, or PR link) and then follow the steps to create an account
3. On the child side, if you use the registration link the org name should automatically appear.
4. You can click "Is this not right?" so you can edit the code or choose "I don't have a code"

## Types of changes

What types of changes does this pull request introduce?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (non-breaking change that does not add functionality but makes code cleaner or more efficient)
- [ ] Documentation Update
- [ ] Tests (new or updated tests)
- [ ] Style (changes to code styling)
- [ ] CI (continuous integration changes)
- [ ] Repository Maintenance
- [ ] Other (please describe below)

## Checklist

- [x] I have read the [guidelines for contributing](https://github.com/yeatmanlab/roar-dashboard/blob/main/.github/CONTRIBUTING.md).
- [x] The changes in this PR are as small as they can be. They represent one and only one fix or enhancement.
- [x] Linting checks pass with my changes.
- [x] Any existing unit tests pass with my changes.
- [x] Any existing end-to-end tests pass with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] If this PR fixes an existing issue, I have added a unit or end-to-end test that will detect if this issue reoccurs.
- [ ] I have added JSDoc comments as appropriate.
- [ ] I have added the necessary documentation to the [roar-docs repository](https://github.com/yeatmanlab/roar-docs).
- [x] I have shared this PR on the roar-pr-reviews channel (if I have access)
- [x] I have linked relevant issues (if any)

### CODE REVIEW

**Review by ksmontville — COMMENTED** [review]
I think that the "Delete Student" and "Add Another Student" buttons should follow the same color/highlighting/focus theme as the ones below.

Currently they looked disabled and then turn red when you hover over the, which is the opposite behavior as the buttons below it.

image

File: src/components/ListOrgs.vue [hunk 1]
```diff
@@ -250,6 +336,23 @@ const tableData = computed(() => {
...
+  const collections = [allDistricts.value, allSchools.value, allClasses.value, allGroups.value];
+
+  for (const collection of collections) {
+    const match = collection?.find((item) => item.id === selectedOrg && item.currentActivationCode);
+    if (match) {
+      activationCode.value = match.currentActivationCode;
+      isDialogVisible.value = true;
+      break;
+    }
+  }
```
Comment (richford) [inline src/components/ListOrgs.vue:349]: I think we can retrieve the activation code without querying all of the orgs:

```suggestion
  const orgInfo = await fetchDocById(activeOrgType.value, orgId);
  if (orgInfo?.currentActivationCode) {
    activationCode.value = orgInfo.currentActivationCode;
    isDialogVisible.value = true;
  }
```

**Review by richford — CHANGES_REQUESTED** [review]

File: src/helpers/query/utils.js [hunk 2]
```diff
@@ -93,15 +96,22 @@ export const exportCsv = (data, filename) => {
...
 };
 
-export const fetchDocById = async (collection, docId, select, db = 'admin') => {
+export const fetchDocById = async (
+  collection,
+  docId,
+  select,
+  db = 'admin',
+  unauthenticated = false,
+  isShallowError = false,
```
[the remark this answers is not in the document]
Comment (Emily-ejag) [reply src/helpers/query/utils.js:105]: done!

File: src/helpers/query/utils.js [hunk 3]
```diff
@@ -114,7 +124,9 @@ export const fetchDocById = async (collection, docId, select, db = 'admin') => {
       };
     })
     .catch((error) => {
-      console.error(error);
+      if (!isShallowError) {
```
[the remark this answers is not in the document]
Comment (Emily-ejag) [reply src/helpers/query/utils.js:127]: done!

File: src/components/ListOrgs.vue [hunk 4]
```diff
@@ -250,6 +314,23 @@ const tableData = computed(() => {
   });
 });
 
+const ShowCode = (selectedOrg) => {
```
[the remark this answers is not in the document]
Comment (Emily-ejag) [reply src/components/ListOrgs.vue:339]: done!

File: src/components/ListOrgs.vue [hunk 5]
```diff
@@ -57,13 +58,37 @@
...
       </PvTabView>
       <AppSpinner v-else />
     </section>
+    <section class="flex mt-8 justify-content-end">
+      <PvDialog
+        v-model:visible="isDialogVisible"
+        header="Invitation link:"
+        :style="{ width: '50rem' }"
+        :position="position"
+        :modal="true"
```
[the remark this answers is not in the document]
Comment (Emily-ejag) [reply src/components/ListOrgs.vue:75]: Yes!

File: src/components/ListOrgs.vue [hunk 6]
```diff
@@ -57,13 +58,37 @@
             sortable
             :loading="isLoading || isFetching"
             :allow-filtering="false"
+            :is-inside-list-orgs="true"
             @export-all="exportAll"
+            @selected-org-id="ShowCode"
```
[the remark this answers is not in the document]
Comment (Emily-ejag) [reply src/components/ListOrgs.vue:63]: done!

File: src/components/ListOrgs.vue (hunk 1 above)
Comment (Emily-ejag) [reply src/components/ListOrgs.vue:349]: Let me try

File: src/components/ListOrgs.vue [hunk 7]
```diff
@@ -57,13 +58,37 @@
...
+      <PvDialog
+        v-model:visible="isDialogVisible"
+        header="Invitation link:"
+        :style="{ width: '50rem' }"
+        :position="position"
+        :modal="true"
+        :draggable="false"
+      >
+        <PvInputGroup>
+          <PvInputText
```
[the remark this answers is not in the document]
Comment (Emily-ejag) [reply src/components/ListOrgs.vue:82]: Yes! I added readonly to the InputTexts

File: src/components/RoarDataTable.vue [hunk 8]
```diff
@@ -332,11 +332,32 @@
               </PvButton>
             </template>
           </PvColumn>
+          <PvColumn
```
Comment (kellyel) [inline src/components/RoarDataTable.vue:335]: Is it possible to abstract this column out into a general-use button column? Actually, I believe I've already implemented a button column, you can check out Listusers.vue on main to see how it's implemented

**Review by kellyel — CHANGES_REQUESTED** [review]

File: src/components/RoarDataTable.vue (hunk 8 above)
Comment (Emily-ejag) [reply src/components/RoarDataTable.vue]: Done!

**Review by kellyel — APPROVED** [review]

**Review by richford — APPROVED** [review]

**Pull request merged**